### PR TITLE
Move SPM paths around

### PR DESCRIPTION
**Pull request by techhat**
Files in `/srv/salt/` are often regularly edited by users or placed under version control (or both). Files provided by SPM are not frequently edited, and should be separated. This will alleviate confusion, and reduce redundancy by keeping package-provided files out of version control.

### CODE REVIEW

**Comment by techhat** [comment]
@s0undt3ch, you should probably review this before we merge.

File: salt/config.py [hunk 1]
```diff
@@ -1197,13 +1201,15 @@
...
+    'pillar_path': '/srv/spm/pillar',
+    'reactor_path': '/srv/spm/reactor',
     'spm_logfile': '/var/log/salt/spm',
     # spm_repos_config also includes a .d/ directory
     'spm_repos_config': '/etc/salt/spm.repos',
     'spm_cache_dir': os.path.join(salt.syspaths.CACHE_DIR, 'spm'),
-    'spm_build_dir': '/srv/spm',
-    'spm_build_exclude': ['.git'],
+    'spm_build_dir': '/srv/spm_build',
+    'spm_build_exclude': ['.git', '.svn'],
```
Comment (s0undt3ch) [inline salt/config.py]: Add `.hg` since you're at it...

File: salt/syspaths.py [hunk 2]
```diff
@@ -104,6 +105,18 @@
...
+    SPM_FORMULA_PATH = os.path.join(SRV_ROOT_DIR, 'spm', 'salt')
+
+SPM_PILLAR_PATH = __generated_syspaths.SPM_PILLAR_PATH
+if SPM_PILLAR_PATH is None:
+    SPM_PILLAR_PATH = os.path.join(SRV_ROOT_DIR, 'spm', 'pillar')
+
+SPM_REACTOR_PATH = __generated_syspaths.SPM_REACTOR_PATH
+if SPM_REACTOR_PATH is None:
+    SPM_REACTOR_PATH = os.path.join(SRV_ROOT_DIR, 'spm', 'reactor')
+
```
Comment (s0undt3ch) [inline salt/syspaths.py]: The above suggest that we support these paths in our `setup.py`, do we?

Comment (s0undt3ch) [inline salt/syspaths.py]: We don't, they should be added...

**Comment by s0undt3ch** [comment]
Aside from my comments, which need to be addressed, it looks good.

File: salt/syspaths.py (hunk 2 above)
Comment (techhat) [inline salt/syspaths.py]: We do need to support it there, since the SPM paths will differ on other operating systems.

File: salt/config.py (hunk 1 above)
Comment (techhat) [inline salt/config.py]: Will do.

**Comment by s0undt3ch** [comment]
I'll submit a PR for the setup.py work, let's merge this now.

**Comment by basepi** [comment]
This is breaking my bootstrapped 2015.8:

```
Traceback (most recent call last):
  File "/usr/bin/salt-minion", line 26, in <module>
    salt_minion()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 106, in salt_minion
    import salt.cli.daemons
  File "/usr/lib/python2.7/dist-packages/salt/cli/daemons.py", line 42, in <module>
    from salt.utils import migrations
  File "/usr/lib/python2.7/dist-packages/salt/utils/migrations.py", line 12, in <module>
    import salt.syspaths as syspaths
  File "/usr/lib/python2.7/dist-packages/salt/syspaths.py", line 108, in <module>
    SPM_FORMULA_PATH = __generated_syspaths.SPM_FORMULA_PATH
AttributeError: 'module' object has no attribute 'SPM_FORMULA_PATH'
```

**Comment by s0undt3ch** [comment]
That's a PR that needs to be done. Ill get that done tonight or first thing in the morning.

**Comment by scoates** [comment]
FWIW, this is also broken for me; same as @basepi.

**Comment by techhat** [comment]
Ref #26723.

**Pull request merged**